File: lib/models/leave.dart
class Leave {
  final String id;
  final String leaveType;
  final String approver;
  final String fromDate;
  final String toDate;
  final String reason;

  Leave({
    required this.id,
    required this.leaveType,
    required this.approver,
    required this.fromDate,
    required this.toDate,
    required this.reason,
  });

  factory Leave.fromJson(Map<String, dynamic> json) {
    return Leave(
      id: json['_id'],
      leaveType: json['leaveType'],
      approver: json['approver'],
      fromDate: json['fromDate'],
      toDate: json['toDate'],
      reason: json['reason'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "leaveType": leaveType,
      "approver": approver,
      "fromDate": fromDate,
      "toDate": toDate,
      "reason": reason,
    };
  }
}

File: backend/models/Leave.js
const mongoose = require('mongoose');

const leaveSchema = new mongoose.Schema({
  employeeId: { type: String, required: true },
  employeeName: { type: String, required: true },
  position: { type: String, required: true },
  leaveType: { type: String, required: true },
  
  approver: { type: String, required: true }, // "admin" | "superadmin"
  applicantRole: { type: String, required: true }, // "employee" | "intern" | "admin"

  fromDate: { type: Date, required: true },
  toDate: { type: Date, required: true },
  reason: { type: String, required: true },
  status: {
    type: String,
    enum: ["Pending", "Approved", "Rejected", "Cancelled"],
    default: "Pending",
  },
  numberOfDays: { type: Number, default: 1 },
  cancelledAt: { type: Date },
}, { timestamps: true });

const Leave = mongoose.models.Leave || mongoose.model('Leave', leaveSchema);
module.exports = Leave;
